common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AzureActiveDirectoryAuthorizationRequest.java

```diff
@@ -22,34 +22,175 @@
+    private String mClaimsChallenge;
+
+    public AzureActiveDirectoryAuthorizationRequest(final String responseType,
+                                                    @NonNull final String clientId,
+                                                    final String redirectUri,
+                                                    final String state,
+                                                    final Set<String> scope,
+                                                    @NonNull final URL authority,
+                                                    @NonNull final String authorizationEndpoint,
+                                                    final String loginHint,
+                                                    final UUID correlationId,
+                                                    final PkceChallenge pkceChallenge,
+                                                    final String extraQueryParam,
+                                                    final String libraryVersion,
+                                                    @NonNull final String resource,
+                                                    final AzureActiveDirectoryPromptBehavior promptBehavior,
+                                                    final String claimsChallenge) {
+        super(responseType, clientId, redirectUri, state, scope, authority, authorizationEndpoint,
+                loginHint, correlationId, pkceChallenge, extraQueryParam, libraryVersion);
+        mResource = resource;
+        mPromptBehavior = promptBehavior;
+        mClaimsChallenge = claimsChallenge;
+    }
+
+    public String getResource() {
+        return mResource;
+    }
+
+    public void setResource(final String resource) {
+        mResource = resource;
+    }
+
+    public AzureActiveDirectoryPromptBehavior getPromptBehavior() {
+        return mPromptBehavior;
+    }
+
+    public void setPromptBehavior(final AzureActiveDirectoryPromptBehavior promptBehavior) {
+        mPromptBehavior = promptBehavior;
+    }
+
+    public String getClaimsChallenge() {
+        return mClaimsChallenge;
+    }
+
+    public void setClaimsChallenge(final String claimsChallenge) {
+        mClaimsChallenge = claimsChallenge;
+    }
+
+    @Override
+    public String getAuthorizationStartUrl() throws UnsupportedEncodingException, ClientException {
+        final String authorizationUrl = StringExtensions.appendQueryParameterToUrl(
+                getAuthorizationEndpoint(),
+                createAuthorizationRequestParameters());
+        Logger.infoPII(TAG, getCorrelationId().toString(), "Request uri to authorize endpoint is: " + authorizationUrl);
+        return authorizationUrl;
+    }
+
+    private void addDiagnosticParameters(@NonNull final Map<String, String> requestParameters) {
+        // append device and platform info in the query parameters
+        requestParameters.put(LIB_ID_PLATFORM, PLATFORM_VALUE);
```

Comment:
I'd rather just serialize the authorization request... rather than constructing it with methods.  I'm fine with checking part in as in and then refactoring.